From CI:

```
Running shellcheck on cli/access-permissions.sh

In cli/access-permissions.sh line 22:
chown -R :www-data $to_update
                   ^--------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
chown -R :www-data "$to_update"


In cli/access-permissions.sh line 25:
chmod -R g+rX $to_update
              ^--------^ SC2086 (info): Double quote to prevent globbing and word splitting.

Did you mean: 
chmod -R g+rX "$to_update"

For more information:
  https://www.shellcheck.net/wiki/SC2086 -- Double quote to prevent globbing ...
```

--- a/Docker/Dockerfile
+++ b/Docker/Dockerfile
@@ -16,7 +16,7 @@ RUN apt-get update && \
 RUN mkdir -p /var/www/FreshRSS/ /run/apache2/
 WORKDIR /var/www/FreshRSS
 
-COPY --chown=:www-data . /var/www/FreshRSS
+COPY --chown=root:www-data . /var/www/FreshRSS
 COPY ./Docker/*.Apache.conf /etc/apache2/sites-available/
 
 ARG FRESHRSS_VERSION

--- a/Docker/Dockerfile-Alpine
+++ b/Docker/Dockerfile-Alpine
@@ -13,7 +13,7 @@ RUN apk add --no-cache \
 RUN mkdir -p /var/www/FreshRSS /run/apache2/
 WORKDIR /var/www/FreshRSS
 
-COPY --chown=:www-data . /var/www/FreshRSS
+COPY --chown=root:www-data . /var/www/FreshRSS
 COPY ./Docker/*.Apache.conf /etc/apache2/conf.d/
 
 ARG FRESHRSS_VERSION

--- a/Docker/Dockerfile-Newest
+++ b/Docker/Dockerfile-Newest
@@ -14,7 +14,7 @@ RUN echo 'http://dl-cdn.alpinelinux.org/alpine/edge/testing' >> /etc/apk/reposit
 RUN mkdir -p /var/www/FreshRSS /run/apache2/
 WORKDIR /var/www/FreshRSS
 
-COPY --chown=:www-data . /var/www/FreshRSS
+COPY --chown=root:www-data . /var/www/FreshRSS
 COPY ./Docker/*.Apache.conf /etc/apache2/conf.d/
 
 ARG FRESHRSS_VERSION

--- a/Docker/Dockerfile-Oldest
+++ b/Docker/Dockerfile-Oldest
@@ -13,7 +13,7 @@ RUN apk add --no-cache \
 RUN mkdir -p /var/www/FreshRSS /run/apache2/
 WORKDIR /var/www/FreshRSS
 
-COPY --chown=:www-data . /var/www/FreshRSS
+COPY --chown=root:www-data . /var/www/FreshRSS
 COPY ./Docker/*.Apache.conf /etc/apache2/conf.d/
 
 ARG FRESHRSS_VERSION

--- a/Docker/entrypoint.sh
+++ b/Docker/entrypoint.sh
@@ -59,6 +59,7 @@ if [ -n "$FRESHRSS_INSTALL" ]; then
 		echo 'ℹ️ FreshRSS already installed; no change performed.'
 	elif [ $EXITCODE -eq 0 ]; then
 		echo '✅ FreshRSS successfully installed.'
+		REAPPLY_PERMISSIONS=:
 	else
 		echo '❌ FreshRSS error during installation!'
 		exit $EXITCODE
@@ -76,12 +77,15 @@ if [ -n "$FRESHRSS_USER" ]; then
 	elif [ $EXITCODE -eq 0 ]; then
 		echo '✅ FreshRSS user successfully created.'
 		./cli/list-users.php | xargs -n1 ./cli/actualize-user.php --user
+		REAPPLY_PERMISSIONS=:
 	else
 		echo '❌ FreshRSS error during the creation of a user!'
 		exit $EXITCODE
 	fi
 fi
 
-./cli/access-permissions.sh --only-userdirs
+if [ -n "$REAPPLY_PERMISSIONS" ]; then
+	./cli/access-permissions.sh --only-userdirs
+fi
 
 exec "$@"

--- a/cli/access-permissions.sh
+++ b/cli/access-permissions.sh
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Apply access permissions
 
 if [ ! -f './constants.php' ] || [ ! -d './cli/' ]; then
@@ -11,19 +11,22 @@ if [ "$(id -u)" -ne 0 ]; then
 	exit 3
 fi
 
-# If specified, only modify the data and extension dirs
+# Always fix permissions on the data and extensions directories
+# Optionally also fix permissions on the entire webapp
+data_path="${DATA_PATH:-./data}"
+to_update=("${data_path}")
 if [ "${1:-}" = "--only-userdirs" ]; then
-	to_update="./data ./extensions"
+	to_update+=("./extensions")
 else
-	to_update="."
+	to_update+=(".")
 fi
 
 # Based on group access
-chown -R :www-data $to_update
+chown -R :www-data "${to_update[@]}"
 
 # Read files, and directory traversal
-chmod -R g+rX $to_update
+chmod -R g+rX "${to_update[@]}"
 
-# Write access
-mkdir -p ./data/users/_/
-chmod -R g+w ./data/
+# Write access to data
+mkdir -p "${data_path}/users/_/"
+chmod -R g+w "${data_path}"

--- a/cli/access-permissions.sh
+++ b/cli/access-permissions.sh
@@ -20,6 +20,8 @@ else
 	to_update="."
 fi
 
+mkdir -p "${data_path}/users/_/"
+
 # Based on group access
 chown -R :www-data "$data_path" "$to_update"
 
@@ -27,5 +29,4 @@ chown -R :www-data "$data_path" "$to_update"
 chmod -R g+rX "$data_path" "$to_update"
 
 # Write access to data
-mkdir -p "${data_path}/users/_/"
 chmod -R g+w "$data_path"

--- a/cli/access-permissions.sh
+++ b/cli/access-permissions.sh
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/bin/sh
 # Apply access permissions
 
 if [ ! -f './constants.php' ] || [ ! -d './cli/' ]; then
@@ -12,21 +12,20 @@ if [ "$(id -u)" -ne 0 ]; then
 fi
 
 # Always fix permissions on the data and extensions directories
-# Optionally also fix permissions on the entire webapp
+# If specified, only fix the data and extensions directories
 data_path="${DATA_PATH:-./data}"
-to_update=("${data_path}")
 if [ "${1:-}" = "--only-userdirs" ]; then
-	to_update+=("./extensions")
+	to_update="./extensions"
 else
-	to_update+=(".")
+	to_update="."
 fi
 
 # Based on group access
-chown -R :www-data "${to_update[@]}"
+chown -R :www-data "$data_path" "$to_update"
 
 # Read files, and directory traversal
-chmod -R g+rX "${to_update[@]}"
+chmod -R g+rX "$data_path" "$to_update"
 
 # Write access to data
 mkdir -p "${data_path}/users/_/"
-chmod -R g+w "${data_path}"
+chmod -R g+w "$data_path"

